File: src/engine/transpile/src/test/resources/golden/SYK006/WS_制御フラグ.java
package cobolinsight.generated;

import cobolinsight.runtime.CobolRuntime;
import java.util.Arrays;

/** SYK006 のデータ部レコード WS-制御フラグ を逐語対訳した自動生成コード。 */
public final class WS_制御フラグ {

    /** 01 WS-制御フラグ 総バイト長 1。 */
    public static final int LENGTH = 1;

    private final byte[] data;

    public WS_制御フラグ() {
        this.data = new byte[1];
    }

    public WS_制御フラグ(byte[] source) {
        this.data = Arrays.copyOf(source, 1);
    }

    public byte[] data() {
        return data;
    }

    // WS-EOF-FLAG
    public String get_WS_EOF_FLAG() {
        return CobolRuntime.decodeAlphanumeric(data, 0, 1);
    }
    public void set_WS_EOF_FLAG(String value) {
        CobolRuntime.encodeAlphanumeric(data, 0, 1, value);
    }

    public boolean is_WS_EOF() {
        return get_WS_EOF_FLAG().equals("Y");
    }
}

File: src/engine/transpile/src/test/resources/golden/SYK001/WS_制御フラグ.py
"""SYK001 のデータ部レコード WS-制御フラグ を逐語対訳した自動生成コード。"""
from cobol_runtime import (
    decode_alphanumeric,
    decode_binary,
    decode_packed,
    decode_zoned,
    encode_alphanumeric,
    encode_binary,
    encode_packed,
    encode_zoned,
)


class WS_制御フラグ:
    """01 WS-制御フラグ 総バイト長 1。"""

    LENGTH = 1

    def __init__(self, data=None):
        if data is None:
            self._data = bytearray(1)
        else:
            self._data = bytearray(data)
            if len(self._data) < 1:
                self._data.extend(b"\x00" * (1 - len(self._data)))

    @property
    def data(self):
        return self._data

    # WS-EOF-FLAG
    def get_WS_EOF_FLAG(self):
        return decode_alphanumeric(self._data, 0, 1)
    def set_WS_EOF_FLAG(self, value):
        encode_alphanumeric(self._data, 0, 1, value)

    def is_WS_EOF(self):
        return self.get_WS_EOF_FLAG() == "Y"
